src/games/strategy/triplea/ai/proAI/util/ProTransportUtils.java

```diff
@@ -162,6 +162,25 @@ public int findUnitsTransportCost(final List<Unit> units) {
     return transportCost;
   }
 
+  public List<Unit> getAirThatCantLandOnCarrier(final PlayerID player, final Territory t, final List<Unit> units) {
+    final GameData data = ai.getGameData();
+    int capacity = AirMovementValidator.carrierCapacity(units, t);
+    final Collection<Unit> airUnits = Match.getMatches(units, ProMatches.unitIsAlliedAir(player, data));
+    List<Unit> airThatCantLand = new ArrayList<Unit>();
+    for (final Unit airUnit : airUnits) {
+      final UnitAttachment ua = UnitAttachment.get(airUnit.getType());
+      final int cost = ua.getCarrierCost();
+      if (cost != -1) {
```

Comment:
Agree that comparing against 0 is probably better but all the existing code related to carrier cost checks not equal to -1 so its better to keep it consistent otherwise it'll be confusing on why some of it checks are against 0 and other code against -1.